examples/ConfigurationStoreBackup/BackupAppConfigurationStore.cs

```diff
@@ -0,0 +1,177 @@
+
+                // If there are any valid App Configuration events, update secondary store.
+                if (updatedKeyLabels.Count > 0)
+                {
+                    bool backupCompleted = await BackupKeyValuesAsync(updatedKeyLabels, log).ConfigureAwait(false);
+                    if (!backupCompleted)
+                    {
+                        // Abort this function without deleting retrievedMessages from storage queue.
+                        log.LogError($"Backup failed because of error(s) returned by Azure App Configuration service.");
+                        return;
+                    }
+                }
+
+                // Delete this batch of events from storage queue.
+                foreach (QueueMessage message in retrievedMessages.Value)
+                {
+                    log.LogDebug($"Deleting message: {message.MessageId}");
+                    await queueClient.DeleteMessageAsync(message.MessageId, message.PopReceipt).ConfigureAwait(false);
+                }
+            }
+        }
+
+        private static async Task<bool> AnyMessageInQueueAsync()
+        {
+            var peekedMessages = await queueClient.PeekMessagesAsync().ConfigureAwait(false);
+            return peekedMessages.Value.Length > 0; 
+        }
+
+        private static HashSet<KeyLabel> ExtractKeyLabelsFromEvents(QueueMessage[] retrievedMessages, ILogger log)
+        {
+            HashSet<KeyLabel> updatedKeyLabels = new HashSet<KeyLabel>(new KeyLabelComparer());
+            foreach (QueueMessage message in retrievedMessages)
+            {
+                string decodedMessage = string.Empty;
+                try
+                {
+                    // Event grid will encode events in Base64 format before publishing to storage queue. Decode the message before parsing it.
+                    decodedMessage = Encoding.UTF8.GetString(Convert.FromBase64String(message.MessageText));
+                    JToken dataObject = JObject.Parse(decodedMessage).Property("data")?.Value;
+                    if (dataObject != null)
+                    {
+                        string key = dataObject["key"].ToString();
+                        string label = dataObject["label"]?.ToString();
+                        updatedKeyLabels.Add(new KeyLabel(key, label));
+                    }
+                }
+                catch (FormatException)
+                {
+                    // If its not in valid base64 format, ignore the queue message. 
+                    log.LogInformation($"Queue message in invalid Base64 format will be ignored.\nMessage: {message.MessageText}");
+                }
+                catch (JsonReaderException)
+                {
+                    // If its not a valid JSON, ignore the queue message. 
+                    log.LogInformation($"Queue message in invalid JSON format will be ignored.\nMessage: {decodedMessage}");
+                }
+                catch (NullReferenceException)
```

Comment:
We shouldn't catch `NullReferenceException` since it doesn't necessarily mean it was thrown due to `key` being `null`. We can add an `if` check for `key` and ignore message if it is `null`.